packages/dapp/src/components/GetStarted.tsx

```diff
@@ -0,0 +1,248 @@
+
+const Wrapper = styled.div`
+  display: flex;
+  justify-content: space-between;
+  margin: 0 auto;
+  max-width: 1000px;
+  padding: 75px 25px 50px;
+  width: 100%;
+
+  ${mediaQueries.MOBILE} {
+    padding: 50px 20px;
+  }
+  ${COLUMN_BREAK_QUERY} {
+    flex-direction: column-reverse;
+  }
+  ${mediaQueries.MOBILE_SMALL} {
+    padding: 50px 0;
+  }
+`;
+const Info = styled.div`
+  max-width: 360px;
+  font-size: 15px;
+  padding: 72px 0 0 24px;
+  line-height: 24px;
+
+  ${COLUMN_BREAK_QUERY} {
+    margin: auto;
+  }
+  ${mediaQueries.MOBILE_SMALL} {
+    max-width: calc(100% - 50px);
+  }
+
+  p {
+    margin: 0 0 36px;
+  }
+`;
+const InfoHeading = styled.div`
+  position: relative;
+  font-weight: 600;
+`;
+const InfoIconWrap = styled.div`
+  position: absolute;
+  left: -36px;
+  svg circle {
+    stroke-width: 2;
+  }
+`;
+const MoreInfoContainer = styled.div`
+  p {
+    margin: 16px 0;
+    font-size: 14px;
+  }
+  ${InfoHeading} {
+    display: inline-block;
+  }
+  ${Arrow} {
+    position: relative;
+    display: inline-block;
+    top: -2px;
+    left: 5px;
+  }
+`;
+const StyledHollowGreenCheck = styled(HollowGreenCheck)`
+  width: 26px;
+  height: 26px;
+  position: relative;
+  left: -1px;
+`;
+const StyledAvatarGenericIcon = styled(AvatarGenericIcon)`
+  width: 24px;
+  height: 24px;
+`;
+const StyledCircleLockIcon = styled(CircleLockIcon)`
+  position: relative;
+  width: 28px;
+  height: 28px;
+  left: -3px;
+  top: -3px;
+`;
+const StyledButton = styled(Button)`
+  width: 100%;
+  padding: 20px 10px;
+  font-weight: bold;
+  text-transform: none;
+  margin: 24px 0;
+`;
+const MoreInfo = styled.div`
+  font-size: 13px;
+  margin-top: 48px;
+  padding-top: 18px;
+  border-top: 1px solid rgb(233, 233, 234);
+`;
+
+const CTA = styled.div`
+  max-width: 400px;
+  flex-grow: 1;
+  text-align: center;
+  font-size: 14px;
+
+  ${COLUMN_BREAK_QUERY} {
+    margin: auto;
+  }
+  ${mediaQueries.MOBILE_SMALL} {
+    max-width: none;
+    width: 100%;
+  }
+`;
+const CTAShadow = styled.div`
+  border-radius: 8px;
+  box-shadow: 0px 15px 35px 0px rgba(0, 0, 0, 0.09);
+`;
+const CTATop = styled.div`
+  padding: 28px 24px 32px;
+`;
+const CTABottom = styled.div`
+  padding: 32px 24px;
+  background: rgb(250, 250, 251);
+`;
+
+const CTAHeader = styled.div`
+  font-size: 24px;
+  font-weight: bold;
+  letter-spacing: 0.25px;
+  line-height: 30px;
+  margin-bottom: 24px;
+`;
+const CTASubhead = styled.div`
+  font-size: 17px;
+  letter-spacing: -0.04px;
+  line-height: 25px;
+`;
+const CTASubheadWelcome = styled(CTASubhead)`
+  font-weight: bold;
+  margin-bottom: 16px;
+`;
+const CTANotice = styled.div`
+  font-size: 13px;
+`;
+
+const GetStartedPage = () => {
+  const context = React.useContext<ICivilContext>(CivilContext);
+  if (!context || !context.auth) {
+    return <LoadingMessage />;
+  }
+  if (context.currentUser) {
+    // @TODO/tobek For logged-in users there's a flash of the page before redirect; there shouldn't be.
```

Comment:
@dankins this is probably because the user is being loaded, right? so for a tiny bit there's no `currentUser`, then the graphql stuff finishes and sets the user in context, right? is there a currentUserLoading?

Reply:
That is exactly it. However, 1) there is a `context.auth.loading` flag I assume is for this use case so I'll add a check for that in, and 2) even without using that, I'm not seeing this flash any more so... that's cool.